ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/snmp-settings.j2

```diff
@@ -105,15 +108,27 @@ snmp-server location {{ snmp_server.location }}
 {%             if user.version is arista.avd.defined %}
 {%                 set user_cli = user_cli ~ " " ~ user.version %}
 {%             endif %}
-{%             if user.auth is arista.avd.defined
+{%             if user.localized is arista.avd.defined
```

Comment:
Missing check for `true`.

Instead of repeating the code, the idea with the `user_cli` var, is that you can just build the parts you need. I would go back to the old code, and then just insert another `if user.localized is arista.avd.defined(true) and user.version is arista.avd.defined('v3') and snmp_server.engineid.local is arista.avd.defined`. The CLI accepts just the `localized <engineid>` CLI, without auth following. So it should not be combined with the checks for auth. You _could_ move the check for v3 out to a wider if block covering both.

Reply:
The CLI does accept it but then it is not part of the show running config:
```
Spine1(config)#snmp-server user blah blahgroup v3 localized 424242424242
Spine1(config)#show run | se blah
snmp-server user blah blahgroup v3
```

Reply:
that being said happy to reorganise the code with a bunch of if statement to insert localized or not but given the behavior I would end up adding the lines:

`if user.localized is arista.avd.defined(true) and user.version is arista.avd.defined('v3') and snmp_server.engineid.local is arista.avd.defined and user.auth is arista.avd.defined and user.auth_passphrase is arista.avd.defined `

Also the previous code could lead to configure priv without auth (even without the `localized` keyword) which is not impossible:
```
Leaf1(config)#snmp-server user test test-group v3 ?
  auth       User authentication settings
  localized  Specify localized auth and privacy passphrases
  <cr>
```

Reply:
If you don't want to output localized alone, you can still just add a `if user.localized is arista.avd.defined(true)` inside the existing `if user.auth ...`. It would make it simpler to add other knobs later.

Reply:
indeed - that would work - I will modify the code